### DISCUSSION TITLE
RefinableDate01= today is not respecting the site timezone

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#3930] [Q&A]
I have a SharePoint Online site collection with these regional settings (Pacific Time): -

![image](https://github.com/user-attachments/assets/4b8aa12b-3467-4698-9ce9-80f2ae6e7292)

And I am using this Search Query Template inside PnP search result web part to get the items which have their RefinableDate01 =today , as follow:-

```
{searchTerms} ContentType:"Carrier"

RefinableDate01=today
```

where the RefinableDate01 is linked to a SharePoint Date/Time field named "PU-ETA". now i am facing this problem:-

1- Currently the date time is 22 August 03:40 am Pacific Time + 10:40 am UTC.

2- I added a list item and i defined the PU-ETA Date/Time as "8/21/2024 10:00 pm ":-

![image](https://github.com/user-attachments/assets/9f90109e-9e22-473c-be19-f54ad64c5282)

3- This will be stored inside SharePoint using UTC which will be "2024-08-22T05:00:00Z".

So when i do a search using this Query Template: -

```
{searchTerms} ContentType:"Carrier"

RefinableDate01=today
```

I will get the above item, although based on the site time zone which the user follow this item is not for today 22 August (based on their timezone) it is for yesterday 21 August.. so how i can fix this? so i can do the check for today based on the site settings timezone? and not based on UTC? so when the user specifies the Date/Time to be PU-ETA as 8/21/2024 10 pm this item should be shown inside the Today report on 21 August Pacific Time ? is this possible?

Thanks

**Comment by wobba** [comment]
Use the token `{today}` instead of the reserved word `today`.  Something like:`RefinableDate01={today}`

**Reply by mvcsharepointdev** [reply]
If i use it as {today} instead of today, i will not get any result:-

![image](https://github.com/user-attachments/assets/d9878ebc-f4b6-4aaf-8711-bd4a5fea50b4)

while if i write it in this way:-

```
{searchTerms} ContentType:"Carrier"
RefinableDate01=today
```

i will get 6 results

**Reply by kasperbolarsen** [reply]
Try to have a look at the query in the Dev tool ( Network tab, search for "query" ) and I would expect that the value for RefinableDate01 doesn't match the value for {today} , there is always something wrong with the time segment that needs adjustment
